src/main/java/internet/shop/dao/impl/DriverDaoImpl.java

```diff
@@ -1,45 +1,117 @@
+        List<Driver> driverList = new ArrayList<>();
+        String query = "SELECT * FROM drivers where driver_deleted = false;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            driverList.add(setDataIntoDriver(resultSet));
+            while (resultSet.next()) {
+                driverList.add(setDataIntoDriver(resultSet));
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't get all data from DB", exception);
+        }
+        return driverList;
     }
 
     @Override
     public Driver update(Driver driver) {
-        List<Driver> dbDriver = Storage.getDrivers();
-        IntStream.range(0, dbDriver.size())
-                .filter(i -> dbDriver.get(i).equals(driver))
-                .findFirst()
-                .ifPresent(i -> dbDriver.set(i, driver));
-        return driver;
+        String query = "UPDATE drivers SET "
+                + "driver_name = ?, driver_license_number = ? "
+                + "where driver_deleted = false and driver_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, driver.getName());
+            preparedStatement.setString(2, driver.getLicenceNumber());
+            preparedStatement.setLong(3, driver.getId());
+
+            if (preparedStatement.executeUpdate() > 0) {
+                return driver;
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't update the data - " + driver, exception);
+        }
+        throw new RuntimeException("No driver " + driver + " to update in DB");
     }
 
     @Override
     public boolean delete(Long id) {
-        return Storage.getDrivers().removeIf(i -> i.getId().equals(id));
+        String query = "UPDATE drivers SET driver_deleted = true "
+                + "where driver_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't delete the data by id: " + id, exception);
+        }
+    }
+
+    private Driver setDataIntoDriver(ResultSet resultSet) throws SQLException {
+        Driver driver = new Driver();
+        if (resultSet.next()) {
+            driver.setId(resultSet.getObject(1, Long.class));
+            driver.setName(resultSet.getObject(2, String.class));
+            driver.setLicenceNumber(resultSet.getObject(3, String.class));
+        }
```

Comment:
```suggestion
        Driver driver = new Driver();
        driver.setId(resultSet.getObject(1, Long.class));
        driver.setName(resultSet.getObject(2, String.class));
        driver.setLicenceNumber(resultSet.getObject(3, String.class));
```
let's pass the non-empty resultSet to here

Reply:
okey